crates/nu-cli/src/env/environment_syncer.rs

```diff
@@ -49,7 +49,9 @@ impl EnvironmentSyncer {
                 if name != "path" && name != "PATH" {
                     // account for new env vars present in the current session
                     // that aren't loaded from config.
-                    environment.add_env(&name, &value);
+                    environment.add_env(&name, &value, false);
+
+                    environment.maintain_directory_environment().ok();
```

Comment:
I can't remember - do you know if this code is run just at startup or between calls. For example, if the user updates the list of allowed directories, do they have to restart Nu for those directories to get picked up?

Reply:
The code snippet you highlighted is run between calls, but it does not update directories. Allowed directories are set when a new environment is created using `from_config()`, which is only run once, by `create_default_context()` in `cli.rs`. (I believe. This is what I was hoping for anyway, and it appears to be the case).
I thought this was fine behavior, as this way `config.toml` only has to be opened and read once at startup, at least for the purposes of this code.